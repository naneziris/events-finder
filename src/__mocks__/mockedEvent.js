export const event = {
    "name": "Cher",
    "type": "event",
    "id": "G5vzZpkJt1Aen",
    "test": false,
    "url": "https://purchase.tickets.com/buy/TicketPurchase?orgid=47569&pid=8781393",
    "locale": "en-us",
    "images": [
      {
        "ratio": "16_9",
        "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_RECOMENDATION_16_9.jpg",
        "width": 100,
        "height": 56,
        "fallback": false
      },
      {
        "ratio": "16_9",
        "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_TABLET_LANDSCAPE_16_9.jpg",
        "width": 1024,
        "height": 576,
        "fallback": false
      },
      {
        "ratio": "16_9",
        "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_RETINA_PORTRAIT_16_9.jpg",
        "width": 640,
        "height": 360,
        "fallback": false
      },
      {
        "ratio": "16_9",
        "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_EVENT_DETAIL_PAGE_16_9.jpg",
        "width": 205,
        "height": 115,
        "fallback": false
      },
      {
        "ratio": "3_2",
        "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_RETINA_PORTRAIT_3_2.jpg",
        "width": 640,
        "height": 427,
        "fallback": false
      },
      {
        "ratio": "16_9",
        "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_RETINA_LANDSCAPE_16_9.jpg",
        "width": 1136,
        "height": 639,
        "fallback": false
      },
      {
        "ratio": "3_2",
        "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_ARTIST_PAGE_3_2.jpg",
        "width": 305,
        "height": 203,
        "fallback": false
      },
      {
        "ratio": "16_9",
        "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_TABLET_LANDSCAPE_LARGE_16_9.jpg",
        "width": 2048,
        "height": 1152,
        "fallback": false
      },
      {
        "ratio": "3_2",
        "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_TABLET_LANDSCAPE_3_2.jpg",
        "width": 1024,
        "height": 683,
        "fallback": false
      },
      {
        "ratio": "4_3",
        "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_CUSTOM.jpg",
        "width": 305,
        "height": 225,
        "fallback": false
      }
    ],
    "sales": {
      "public": {
        "startDateTime": "2020-03-09T18:01:00Z",
        "startTBD": false,
        "endDateTime": "2020-10-07T02:00:00Z"
      }
    },
    "dates": {
      "start": {
        "localDate": "2020-10-06",
        "localTime": "20:00:00",
        "dateTime": "2020-10-07T02:00:00Z",
        "dateTBD": false,
        "dateTBA": false,
        "timeTBA": false,
        "noSpecificTime": false
      },
      "timezone": "America/Denver",
      "status": {
        "code": "rescheduled"
      },
      "spanMultipleDays": false
    },
    "classifications": [
      {
        "primary": true,
        "segment": {
          "id": "KZFzniwnSyZfZ7v7nJ",
          "name": "Music"
        },
        "genre": {
          "id": "KnvZfZ7vAeA",
          "name": "Rock"
        },
        "subGenre": {
          "id": "KZazBEonSMnZfZ7v6F1",
          "name": "Pop"
        },
        "type": {
          "id": "KZAyXgnZfZ7v7nI",
          "name": "Undefined"
        },
        "subType": {
          "id": "KZFzBErXgnZfZ7v7lJ",
          "name": "Undefined"
        },
        "family": false
      }
    ],
    "promoter": {
      "id": "653",
      "name": "LIVE NATION MUSIC",
      "description": "LIVE NATION MUSIC / NTL / USA"
    },
    "promoters": [
      {
        "id": "653",
        "name": "LIVE NATION MUSIC",
        "description": "LIVE NATION MUSIC / NTL / USA"
      },
      {
        "id": "4576",
        "name": "LIVE NATION 3PE EVENT",
        "description": "LIVE NATION 3PE EVENT / NTL / USA"
      }
    ],
    "ageRestrictions": {
      "legalAgeEnforced": false
    },
    "_links": {
      "self": {
        "href": "/discovery/v2/events/G5vzZpkJt1Aen?locale=en-us"
      },
      "attractions": [
        {
          "href": "/discovery/v2/attractions/K8vZ9171hr7?locale=en-us"
        }
      ],
      "venues": [
        {
          "href": "/discovery/v2/venues/KovZpZA1EanA?locale=en-us"
        }
      ]
    },
    "_embedded": {
      "venues": [
        {
          "name": "First Interstate Arena",
          "type": "venue",
          "id": "KovZpZA1EanA",
          "test": false,
          "url": "https://www.ticketmaster.com/first-interstate-arena-tickets-billings/venue/246706",
          "locale": "en-us",
          "postalCode": "59101",
          "timezone": "America/Denver",
          "city": {
            "name": "Billings"
          },
          "state": {
            "name": "Montana",
            "stateCode": "MT"
          },
          "country": {
            "name": "United States Of America",
            "countryCode": "US"
          },
          "address": {
            "line1": "308 6th Ave N"
          },
          "location": {
            "longitude": "-108.4819363",
            "latitude": "45.7990694"
          },
          "markets": [
            {
              "name": " Montana & Idaho\"",
              "id": "45"
            }
          ],
          "dmas": [
            {
              "id": 229
            },
            {
              "id": 239
            },
            {
              "id": 284
            },
            {
              "id": 287
            },
            {
              "id": 298
            },
            {
              "id": 338
            },
            {
              "id": 378
            }
          ],
          "upcomingEvents": {
            "_total": 5,
            "tmr": 3,
            "ticketmaster": 2
          },
          "_links": {
            "self": {
              "href": "/discovery/v2/venues/KovZpZA1EanA?locale=en-us"
            }
          }
        }
      ],
      "attractions": [
        {
          "name": "Cher",
          "type": "attraction",
          "id": "K8vZ9171hr7",
          "test": false,
          "url": "https://www.ticketmaster.com/cher-tickets/artist/734742",
          "locale": "en-us",
          "externalLinks": {
            "youtube": [
              {
                "url": "https://www.youtube.com/user/cher"
              }
            ],
            "twitter": [
              {
                "url": "https://twitter.com/cher"
              }
            ],
            "lastfm": [
              {
                "url": "http://www.last.fm/music/Cher"
              }
            ],
            "facebook": [
              {
                "url": "https://www.facebook.com/cher"
              }
            ],
            "wiki": [
              {
                "url": "https://en.wikipedia.org/wiki/Cher"
              }
            ],
            "musicbrainz": [
              {
                "id": "bfcc6d75-a6a5-4bc6-8282-47aec8531818"
              }
            ],
            "homepage": [
              {
                "url": "http://www.cher.com/"
              }
            ]
          },
          "images": [
            {
              "ratio": "16_9",
              "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_RECOMENDATION_16_9.jpg",
              "width": 100,
              "height": 56,
              "fallback": false
            },
            {
              "ratio": "16_9",
              "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_TABLET_LANDSCAPE_16_9.jpg",
              "width": 1024,
              "height": 576,
              "fallback": false
            },
            {
              "ratio": "16_9",
              "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_RETINA_PORTRAIT_16_9.jpg",
              "width": 640,
              "height": 360,
              "fallback": false
            },
            {
              "ratio": "16_9",
              "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_EVENT_DETAIL_PAGE_16_9.jpg",
              "width": 205,
              "height": 115,
              "fallback": false
            },
            {
              "ratio": "3_2",
              "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_RETINA_PORTRAIT_3_2.jpg",
              "width": 640,
              "height": 427,
              "fallback": false
            },
            {
              "ratio": "16_9",
              "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_RETINA_LANDSCAPE_16_9.jpg",
              "width": 1136,
              "height": 639,
              "fallback": false
            },
            {
              "ratio": "3_2",
              "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_ARTIST_PAGE_3_2.jpg",
              "width": 305,
              "height": 203,
              "fallback": false
            },
            {
              "ratio": "16_9",
              "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_TABLET_LANDSCAPE_LARGE_16_9.jpg",
              "width": 2048,
              "height": 1152,
              "fallback": false
            },
            {
              "ratio": "3_2",
              "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_TABLET_LANDSCAPE_3_2.jpg",
              "width": 1024,
              "height": 683,
              "fallback": false
            },
            {
              "ratio": "4_3",
              "url": "https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_CUSTOM.jpg",
              "width": 305,
              "height": 225,
              "fallback": false
            }
          ],
          "classifications": [
            {
              "primary": true,
              "segment": {
                "id": "KZFzniwnSyZfZ7v7nJ",
                "name": "Music"
              },
              "genre": {
                "id": "KnvZfZ7vAeA",
                "name": "Rock"
              },
              "subGenre": {
                "id": "KZazBEonSMnZfZ7v6F1",
                "name": "Pop"
              },
              "type": {
                "id": "KZAyXgnZfZ7v7nI",
                "name": "Undefined"
              },
              "subType": {
                "id": "KZFzBErXgnZfZ7v7lJ",
                "name": "Undefined"
              },
              "family": false
            }
          ],
          "upcomingEvents": {
            "_total": 55,
            "tmr": 10,
            "ticketmaster": 45
          },
          "_links": {
            "self": {
              "href": "/discovery/v2/attractions/K8vZ9171hr7?locale=en-us"
            }
          }
        }
      ]
    }
  };
