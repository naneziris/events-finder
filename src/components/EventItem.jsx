import React, { useState } from 'react';
import _filter from 'lodash/filter';
import './EventsList.css';

const EventItem = ({event}) => {
    const { name, id, url, images, dates, classifications, promoter } = event;
    const [showMore, setShowMore] = useState(false);
    const CTAlabel = !showMore ? 'More Info' : 'Hide Info';
    const eventOpenExtraClass = showMore ? "open" : '';

    const eventImage = _filter(images, {width: 640, height: 360})[0].url;
    const eventDate = dates.start.localDate;
    const { segment, genre, subGenre } = classifications[0];
    return (
        <li className={`event ${eventOpenExtraClass}`}>
            <div className="event__container">
                <div className="event__media">
                    <img src={eventImage} alt={name} />
                </div>
                <div className="event__body">
                    <h2 className="event__title">{name}</h2>
                    <h5 className="event__date">Date: {eventDate}</h5>
                    <button className="event__btn" type="button" data-id={id} onClick={()=> setShowMore(!showMore)}>{CTAlabel}</button>
                </div>
            </div>
            {showMore &&(
                <div className="event__footer">
                    <div>
                        <h4>Event website</h4>
                        <a href={url} target="_blank" rel="noopener noreferrer" >Click here</a>
                    </div>
                    <div>
                        <h4>Event type:</h4>
                        <span>{segment.name} ({genre.name}, {subGenre.name})</span>
                    </div>
                    <div>
                        <h4>Promoter:</h4>
                        <div>{promoter.name}<br />{promoter.description}</div>
                    </div>
                    <button className="event__btn event__btn--small" type="button" data-id={id} onClick={()=> setShowMore(!showMore)}>{CTAlabel}</button>

                </div>
            )}
        </li>
    );
};

export default EventItem;
