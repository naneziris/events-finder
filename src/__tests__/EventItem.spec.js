import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import "../configTests";
import EventItem from '../components/EventItem';
import { event } from '../__mocks__/mockedEvent';

describe('EventItem component', () => {
    it('Renders without a crash', () => {
        const wrapper = shallow(<EventItem event={event} />);
        expect(wrapper).toBeDefined();
    });
    it('Should match the snapshot', () => {
        const tree = renderer.create(<EventItem  event={event} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('Should show footer when "More info" gets clicked', () => {
        const wrapper = shallow(<EventItem event={event} />);
        expect(wrapper.find('.event__footer').exists()).toBeFalsy();
        wrapper.find('button').simulate('click');
        expect(wrapper.find('.event__footer').exists()).toBeTruthy();
    })
    it('Should hide footer when "Hide Info" gets clicked', () => {
        const wrapper = shallow(<EventItem event={event} />);
        wrapper.find('button').simulate('click');
        wrapper.find('.event__footer').find('button').simulate('click');
        expect(wrapper.find('.event__footer').exists()).toBeFalsy();
    })
})
