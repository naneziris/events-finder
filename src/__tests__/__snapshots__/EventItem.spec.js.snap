// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventItem component Should match the snapshot 1`] = `
<li
  className="event "
>
  <div
    className="event__container"
  >
    <div
      className="event__media"
    >
      <img
        alt="Cher"
        src="https://s1.ticketm.net/dam/a/aa1/03bb556a-482a-4c3a-93e0-2b0412ecbaa1_1198501_RETINA_PORTRAIT_16_9.jpg"
      />
    </div>
    <div
      className="event__body"
    >
      <h2
        className="event__title"
      >
        Cher
      </h2>
      <h5
        className="event__date"
      >
        Date: 
        2020-10-06
      </h5>
      <button
        className="event__btn"
        data-id="G5vzZpkJt1Aen"
        onClick={[Function]}
        type="button"
      >
        More Info
      </button>
    </div>
  </div>
</li>
`;
